client/rpc/ws_client.go

```diff
@@ -673,9 +681,9 @@ func (c *WSClient) dial() error {
 func (c *WSClient) reconnect() error {
 	attempt := 0
 
-	c.reconnecting = true
```

Comment:
why don't we protect the change of `dialing` in this fuction via waitgroup?

Reply:
According to the logic:  We change `dialing` only in `reconnectRoutine` and `dialRoutine`, which can't happen in the same time, there is no concurrent write of `dialing`, I think there is no need to protect `dialing`.